app-laptop/tiny-dfr/tiny-dfr-0.1.1.ebuild

```diff
@@ -0,0 +1,282 @@
+phf_macros-0.10.0
+phf_shared-0.8.0
+phf_shared-0.10.0
+pin-project-lite-0.2.9
+pin-utils-0.1.0
+pkg-config-0.3.27
+ppv-lite86-0.2.17
+precomputed-hash-0.1.1
+privdrop-0.5.3
+proc-macro-crate-1.3.1
+proc-macro-error-1.0.4
+proc-macro-error-attr-1.0.4
+proc-macro-hack-0.5.20+deprecated
+proc-macro2-1.0.60
+quote-1.0.28
+rand-0.7.3
+rand-0.8.5
+rand_chacha-0.2.2
+rand_chacha-0.3.1
+rand_core-0.5.1
+rand_core-0.6.4
+rand_hc-0.2.0
+rand_pcg-0.2.1
+rawpointer-0.2.1
+rayon-1.7.0
+rayon-core-1.11.0
+rctree-0.5.0
+redox_syscall-0.3.5
+regex-1.8.4
+regex-syntax-0.7.2
+rgb-0.8.36
+rustc_version-0.4.0
+rustix-0.37.20
+safe_arch-0.7.0
+scopeguard-1.1.0
+selectors-0.24.0
+semver-1.0.17
+serde-1.0.164
+serde_spanned-0.6.2
+servo_arc-0.2.0
+simba-0.8.1
+siphasher-0.3.10
+slab-0.4.8
+smallvec-1.10.0
+stable_deref_trait-1.2.0
+static_assertions-1.1.0
+string_cache-0.8.7
+string_cache_codegen-0.5.2
+strsim-0.10.0
+syn-1.0.109
+syn-2.0.18
+system-deps-6.1.0
+target-lexicon-0.12.7
+tendril-0.4.3
+thiserror-1.0.40
+thiserror-impl-1.0.40
+tinyvec-1.6.0
+tinyvec_macros-0.1.1
+toml-0.7.4
+toml_datetime-0.6.2
+toml_edit-0.19.10
+typenum-1.16.0
+udev-0.7.0
+unicode-bidi-0.3.13
+unicode-ident-1.0.9
+unicode-normalization-0.1.22
+url-2.4.0
+utf-8-0.7.6
+utf8parse-0.2.1
+version-compare-0.1.1
+version_check-0.9.4
+wasi-0.9.0+wasi-snapshot-preview1
+wasi-0.11.0+wasi-snapshot-preview1
+wasm-bindgen-0.2.87
+wasm-bindgen-backend-0.2.87
+wasm-bindgen-macro-0.2.87
+wasm-bindgen-macro-support-0.2.87
+wasm-bindgen-shared-0.2.87
+wide-0.7.10
+winapi-0.3.9
+winapi-i686-pc-windows-gnu-0.4.0
+winapi-x86_64-pc-windows-gnu-0.4.0
+windows-0.48.0
+windows-sys-0.48.0
+windows-targets-0.48.0
+windows_aarch64_gnullvm-0.48.0
+windows_aarch64_msvc-0.48.0
+windows_i686_gnu-0.48.0
+windows_i686_msvc-0.48.0
+windows_x86_64_gnu-0.48.0
+windows_x86_64_gnullvm-0.48.0
+windows_x86_64_msvc-0.48.0
+winnow-0.4.7
+xml5ever-0.17.0
+"
+
+LIBRSVG_COMMIT="b831e077174ae608d8cd09e532fc0e7ce1fe5c4f"
+
+declare -A GIT_CRATES=(
+	[librsvg]="https://gitlab.gnome.org/GNOME/librsvg;${LIBRSVG_COMMIT}"
+)
+
+inherit cargo udev systemd linux-info
+
+DESCRIPTION="The most basic dynamic function row daemon possible"
+HOMEPAGE="https://github.com/WhatAmISupposedToPutHere/tiny-dfr"
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~arm64"
+
+SRC_URI="
+	$(cargo_crate_uris)
+	https://gitlab.gnome.org/GNOME/librsvg/-/archive/2.56.0/librsvg-2.56.0.tar.gz -> librsvg-2.56.0.crate
+	https://github.com/WhatAmISupposedToPutHere/tiny-dfr/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+"
+
+DEPEND="
+	dev-libs/libinput
+	x11-libs/pango
+"
+
+RDEPEND="${DEPEND}"
+
+pkg_pretend() {
+	local CONFIG_CHECK="~INPUT_UINPUT"
+	[[ ${MERGE_TYPE} != buildonly ]] && check_extra_config
+}
+
+src_install() {
+	cargo_src_install
+
+	insinto /usr/share/tiny-dfr
+	doins share/tiny-dfr/*
+
+	udev_dorules etc/udev/rules.d/*
+	systemd_dounit etc/systemd/system/tiny-dfr.service
+	systemd_newunit /dev/null 'systemd-backlight@backlight:228200000.display-pipe.0.service'
```

Comment:
Does this have any advantage? Shouldn't that be automatically created when the backlight device is present? My plan (for Asahi) was to add tiny-dfr as dependency of the asahi-meta package so it's always present. Since that installs it on devices without touchbar it would be nice to install just the bare minimum.

Reply:
I saw some flakiness with systemd-backlight trying to overtake the backlight control and breaking dfr. Maybe not required but not sure. And it does not hurt touchbar-less devices, since they dont have this service instantiated